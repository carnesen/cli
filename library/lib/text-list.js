"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextList = void 0;
const util_1 = require("./util");
function TextListParagraph(item, targetNameLength = 0) {
    const { name, text } = item;
    const lines = util_1.regularizeText(text).split('\n');
    const paddedName = name.padEnd(targetNameLength);
    let firstLine = paddedName;
    if (lines[0]) {
        firstLine += ` : ${lines[0]}`;
    }
    const padding = ''.padEnd(targetNameLength + 3);
    const paddedLines = lines.slice(1).map((line) => `${padding}${line}`);
    return [firstLine, ...paddedLines].join('\n');
}
function TextList(...items) {
    const targetNameLength = Math.max(...items.map(({ name }) => name.length));
    const paragraphs = items.map((item) => TextListParagraph(item, targetNameLength));
    return paragraphs.join('\n');
}
exports.TextList = TextList;
//# sourceMappingURL=text-list.js.map