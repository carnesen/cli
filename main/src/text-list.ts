import { regularizeText } from './util';

type Item = {
  name: string;
  text?: string;
};

function TextListParagraph(item: Item, targetNameLength = 0): string {
  const { name, text } = item;
  const lines = regularizeText(text).split('\n');
  const paddedName = name.padEnd(targetNameLength);
  let firstLine = paddedName;
  if (lines[0]) {
    firstLine += ` : ${lines[0]}`;
  }
  const padding = ''.padEnd(targetNameLength + 3);
  const paddedLines = lines.slice(1).map((line) => `${padding}${line}`);
  return [firstLine, ...paddedLines].join('\n');
}

export function TextList(...items: Item[]): string {
  const targetNameLength = Math.max(...items.map(({ name }) => name.length));
  const paragraphs = items.map((item) => TextListParagraph(item, targetNameLength));
  return paragraphs.join('\n');
}
