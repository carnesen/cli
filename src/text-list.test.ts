import { TextList } from './text-list';

describe(TextList.name, () => {
  it('Creates a usage string for a branch', () => {
    const usageString = TextList(
      { name: 'foo', text: 'hi\n   bye' },
      { name: 'barrio', text: 'bye' },
    );
    expect(usageString).toMatchSnapshot();
  });
});
