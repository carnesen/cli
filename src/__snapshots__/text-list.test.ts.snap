// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextList Creates a usage string for a branch 1`] = `
"foo    : hi
            bye
barrio : bye"
`;
